crates/core/src/contract/storages/redb.rs

```diff
@@ -0,0 +1,88 @@
+use std::path::Path;
+
+use freenet_stdlib::prelude::*;
+use redb::{Database, TableDefinition};
+
+use crate::wasm_runtime::StateStorage;
+
+const TABLE: TableDefinition<&[u8], &[u8]> = TableDefinition::new("contract");
+
+pub struct ReDb(Database);
+
+impl ReDb {
+    pub async fn new(db_path: &Path) -> Result<Self, redb::Error> {
+        tracing::info!("loading contract store from {db_path:?}");
+
+        Database::create(db_path).map(Self).map_err(Into::into)
+    }
+}
+
+impl ReDb {
+    const STATE_SUFFIX: &'static [u8] = "_key".as_bytes();
+    const PARAMS_SUFFIX: &'static [u8] = "_params".as_bytes();
+}
+
+impl StateStorage for ReDb {
+    type Error = redb::Error;
+
+    async fn store(&mut self, key: ContractKey, state: WrappedState) -> Result<(), Self::Error> {
+        let txn = self.0.begin_write()?;
+
+        {
+            let mut tbl = txn.open_table(TABLE)?;
```

Comment:
why not have 2 different tables one for params and an other for contracts instead